### PR TITLE
feat(all-clouds): created public images subproject and Vagrant documentation

### PR DESCRIPTION
**Pull request by mstpn**
Public images not covered by existing public cloud providers needed a home.
Vagrant documentation added to support users creating and running their own boxes after Ubuntu dropped support in 24.04+.

### CODE REVIEW

File: all-clouds/public-images/index.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
+Ubuntu public images
+====================
+
+**Canonical produce generic (generic kernel) cloud images, LXD images (rootfs tarballs) and KVM optimised cloud images (KVM kernel)**. These images are public (unlike other cloud-specific images) and are available on `cloud-images.ubuntu.com <http://cloud-images.ubuntu.com>`_.
```
Comment (j5awry) [inline all-clouds/public-images/index.rst:4]: leave for now, but the KVM kernel is deprecated. I've got docs in the works (that i'm staring at now)

File: all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+     -device virtio-net-pci,netdev=net00 \
+     -drive if=virtio,format=qcow2,file=./jammy-server-cloudimg-amd64.img \
+     -drive if=virtio,format=raw,file=seed.img
+
+.. note::
+   To exit the QEMU shell, press ``C-a x``
+
+Configure the environment
+-------------------------
+All the commands can be run in an ssh session except ``lb build`` that **has to be run via the serial console**.
```
Comment (j5awry) [inline all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst:112]: this is interesting, and might warrant a small "why"

File: all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+   export SUITE=jammy
+   export ARCH=amd64
+
+   # Defines the binary hooks that will be run.
+   # eg. azure, ec2, qcow2, oracle, etc...
+   # it is a comma separated list of strings
+   export IMAGE_TARGETS=vagrant
+
+   # this is the serial that will be set in /etc/cloud/build.info
+   export NOW=20240409
```
Comment (j5awry) [inline all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst:152]: maybe NOW=$(date +%Y%m%d)

that way they don't have to change it every single date. but :shrug:

File: all-clouds/public-images/public-images-how-to/run-a-vagrant-box.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+The Vagrantfile
+---------------
+Vagrantfiles are written in Ruby. This is handy to know, as you do have access to the core Ruby language. Running ``vagrant init`` provides a fairly extensive, commented file for review. If you need something smaller to start, you can copy the template below to spin up a vm:
+
+.. code:: ruby
+
+   # -*- mode: ruby -*-
+   # vi: set ft=ruby :
+   Vagrant.configure("2") do |config|
+     config.vm.box = "ubuntu/groovy64"
```
Comment (j5awry) [inline all-clouds/public-images/public-images-how-to/run-a-vagrant-box.rst:34]: switch to ubuntu/jammy64

**Review by j5awry — CHANGES_REQUESTED** [review]
Just one change -- point people to jammy64 not groovy. groovy is long gone :)

File: all-clouds/public-images/index.rst (hunk 1 above)
Comment (mstpn) [reply all-clouds/public-images/index.rst:4]: Noted.

File: all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst (hunk 3 above)
Comment (mstpn) [reply all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst:152]: Good idea making it more generic.

File: all-clouds/public-images/public-images-how-to/run-a-vagrant-box.rst (hunk 4 above)
Comment (mstpn) [reply all-clouds/public-images/public-images-how-to/run-a-vagrant-box.rst:34]: Great catch, done.

File: all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst (hunk 2 above)
Comment (mstpn) [reply all-clouds/public-images/public-images-how-to/build-vagrant-with-live-build.rst:112]: Removed that document page altogether.

**Review by j5awry — APPROVED** [review]

**Review by k-dimple — APPROVED** [review]
This LGTM. There are a few minor nits, that I'll address. But before doing that, I'll create a separate project for this and publish it in its own URL.

**Comment by k-dimple** [comment]
@mstpn on second thoughts, could you move your folder out of public-cloud and make it an independent one like the other clouds? Then we can merge this PR and keep the correct git history. I'll create the corresponding project of RTD and the URL.

**Comment by mstpn** [comment]
@k-dimple Built out "Public Images" as a separate project and made associated changes. 

All tests passing locally except for 1 warning on `linkcheck` for the non-existent RTD URL. All check failures above are related to this as well: `loading intersphinx inventory from https://canonical-public-images.readthedocs-hosted.com/en/latest/objects.inv...`.

Will any changes need to be made to add `docs/readthedocs.com:canonical-public-images` to the list of automated checks or is that handled automatically?

**Comment by k-dimple** [comment]
@mstpn Thanks for all the changes! I have merged it and created a new project in readthedocs.

> Will any changes need to be made to add docs/readthedocs.com:canonical-public-images to the list of automated checks or is that handled automatically?

This is handled during creation of the new project.

It is now published at: https://documentation.ubuntu.com/public-images/en/latest/  (and also at https://canonical-public-images.readthedocs-hosted.com/en/latest/)

**Pull request merged**